force-app/main/default/flows/Before_Insert_Update_Public_Event.flow-meta.xml

```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Flow xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>54.0</apiVersion>
+    <description>Create unique Id and set Event Id with it</description>
+    <formulas>
+        <name>CreatedEventId</name>
+        <dataType>String</dataType>
+        <expression>LOWER({!$Record.Name})+LOWER({!$Record.Country__c})+LOWER({!$Record.City__c})+TEXT(DATEVALUE({!$Record.Start_Date__c}))+TEXT(DATEVALUE({!$Record.End_Date__c}))</expression>
```

Comment:
@t-gul As I remember we have decided to split only the date fields into fields. Name, Country, City should be matched seperately in duplicate/matching rule. I was expecting date fields only within this pull request